### ISSUE TITLE
Add an operator for receiving video metadata

### ISSUE THREAD

**Issue by treasan** [NVIDIA/DALI#5630]
### Is this a new feature, an improvement, or a change to existing functionality?

New Feature

### How would you describe the priority of this feature request

Should have (e.g. Adoption is possible, but the performance shortcomings make the solution inferior).

### Please provide a clear description of problem this feature solves

The sample rate (fps) of videos may very and hence the time period a fixed number of frames represent also varies. Having access to either the fps, duration or even the concrete timesteps of each frame is often crucial in many tasks where the actual duration in seconds is more important than the number of frames. For example, I am decoding raw video bytes from a web dataset using the experimental video decoder and I am forced to retreat to other libraries that can give me this kind of information from the raw video bytes (specifically, pytorch's VideoReader API).

### Feature Description

As a user I want to be able to extract information about the sample rate of a video alongside its decoded frames.

### Describe your ideal solution

A new DALI operator that extracts the desired metadata from raw video bytes.
An example video decoding pipeline reading from a webdataset (raw video bytes could also come from an external source):
```
@pipeline_def
def pipeline(tar_paths):
    raw_video = fn.readers.webdataset(tar_paths, ...)
    duration, fps = fn.get_video_metadata(...)
    video = fn.experimental.decoders.video(raw_video)
    return video, duration, fps
```

### Check for duplicates

- [x] I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report

**Comment by JanuszL** [comment]
Hi @treasan,

Thank you for reaching out. Yes, that sounds like a good feature to add. Let us add this to our ToDo list.
Could you also tell me how do you want to utilize this data further? To drive transformations or to feed the model?

**Comment by treasan** [comment]
Hey @JanuszL 

I am training a model, which expects video snippets within a certain duration (in seconds). Furthermore it expects a timestep for each frame, which is used for a temporal positional encoding.

**Comment by treasan** [comment]
Please have a look at another issue/question I have submitted #5626. I explain my pipeline there in more detail.

**tl;dr:**
1. DALI pipeline: Loading raw video bytes from webdataset in a DALI pipeline
2. Python function: Peeking duration and fps metadata from raw video bytes and filter out unwanted videos beforehand (e.g. too short ones)
3. DALI pipeline: Get raw video bytes, duration, fps from external source --> decode video --> return decoded video, duration, fps
4. Generating individual snippets of certain durations, calculate frame timesteps based on duration/fps info, and feed snippet batches to model.

So, optimal for my use-case would be a DALI operator that can peek this metadata information from the raw video bytes already, as I am then able to filter them out before the decoding step (more efficient).
